### PR TITLE
#161342843 Add create comment and create reply routes

### PR DESCRIPTION
**Pull request by TheoOkafor**
#### What does this PR do?
- Add create-a-comment and create-reply functionality

#### Description of Task to be completed?
- Add create comment route
- Add create reply route
- Add tests for both routes

#### How should this be manually tested?
- git clone `https://github.com/andela/haven-ah-backend.git`.
- Install `node` if not installed.
- ensure that PostgreSQL is running on your local machine.
- run `npm install` to install the npm packages.
- run `npm test` to run the unit and integration tests.

To test on postman
- run `npm run start-dev`
- *For Comment,* On postman run POST `localhost:5000/api/articles/:slug/comments` (set JWT token as`x-access-token` in header).
- *For Reply,* On postman run POST `localhost:5000/api/articles/:slug/comments/:parentId` (set JWT token as`x-access-token` in header).

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
#161342843

##### POST Comments

Good Response:

screen shot 2018-11-05 at 5 28 18 pm

Bad Request Response (when body is not included):

screen shot 2018-11-05 at 5 32 17 pm

##### POST Reply
Good Response:

screen shot 2018-11-05 at 5 27 46 pm

Bad Request Response (when parent comment could not be found):

screen shot 2018-11-05 at 5 34 03 pm

### CODE REVIEW

File: server/controllers/article.js [hunk 1]
```diff
@@ -55,6 +55,105 @@ class Article {
       return badHttpResponse(response, 500, 'There was an internal server error', error);
     }
   }
+
+  /** Create a new comment
+   * @param {object} request Request Object
+   * @param {object} response Response Object
+   * @returns {object} Comment Object
+   */
+  static async createComment(request, response) {
```
Comment (victor-nweze-deel) [inline server/controllers/article.js]: We should have this in a comment class for the separation of concerns

Comment (TheoOkafor) [reply server/controllers/article.js]: I see. Thank you.

**Comment by PriscillaSam** [comment]
Nice work @TheoOkafor. 👍

**Comment by ucheg6** [comment]
This looks clean. Good work Theo

**Comment by TheoOkafor** [comment]
Thanks guys and Priscilla.

**Review by wiztemple — COMMENTED** [review]
nice work.

File: server/controllers/comment.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+class Comment {
+  /** Create a new comment
+   * @param {object} request Request Object
+   * @param {object} response Response Object
+   * @returns {object} Comment Object
+   */
+  static async createComment(request, response) {
+    try {
+      const { slug } = request.params;
+      const article = await articleRepo.getArticle(slug);
```
Comment (seyi-adeleke) [inline server/controllers/comment.js]: Isnt there a getArticleBySlug method now?

File: server/controllers/comment.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      if (article === null) {
+        return badHttpResponse(response, 404, 'We could not find this article');
+      }
+
+      request.body.articleId = article.id;
+      const newComment = await commentRepo.createComment(request.body);
+      return goodHttpResponse(response, 201, 'Comment created', newComment);
+    } catch (error) {
```
Comment (seyi-adeleke) [inline server/controllers/comment.js:25]: Probably not a good Idea to return a raw error straight to the client. We should talk about this

File: server/repository/articleRepository.js [hunk 4]
```diff
@@ -73,6 +73,23 @@ class ArticleRepository {
     }
     return singleArticle;
   }
+
+  /** Function to get an article in the database
+   * @param {object} slug object
+   * @returns {object} article object
+   ** otherwise it throws an error
+   */
+  static async getArticle(slug) {
```
Comment (seyi-adeleke) [inline server/repository/articleRepository.js]: see my first comment

**Review by seyi-adeleke — APPROVED** [review]

File: server/controllers/comment.js (hunk 2 above)
Comment (TheoOkafor) [reply server/controllers/comment.js]: I just confirmed that there is. Thanks

File: server/controllers/comment.js (hunk 3 above)
Comment (TheoOkafor) [reply server/controllers/comment.js:25]: Okay

**Pull request merged**